packages/advanced-logic/test/extensions/payment-network/meta.test.ts

```diff
@@ -0,0 +1,385 @@
+        ][0].acceptedTokens = [];
+        // 'must throw'
+        expect(() => {
+          metaPn.applyActionToExtension(
+            MetaCreate.requestStateNoExtensions.extensions,
+            actionWithInvalidToken,
+            MetaCreate.requestStateNoExtensions,
+            TestData.otherIdRaw.identity,
+            TestData.arbitraryTimestamp,
+          );
+        }).toThrowError('acceptedTokens is required');
+      });
+
+      it('cannot applyActionToExtensions of creation with token address not valid', () => {
+        const actionWithInvalidToken = deepCopy(MetaCreate.actionCreationFull);
+        actionWithInvalidToken.parameters[
+          ExtensionTypes.PAYMENT_NETWORK_ID.ANY_TO_ERC20_PROXY
+        ][0].acceptedTokens = ['invalid address'];
+        // 'must throw'
+        expect(() => {
+          metaPn.applyActionToExtension(
+            MetaCreate.requestStateNoExtensions.extensions,
+            actionWithInvalidToken,
+            MetaCreate.requestStateNoExtensions,
+            TestData.otherIdRaw.identity,
+            TestData.arbitraryTimestamp,
+          );
+        }).toThrowError('acceptedTokens must contains only valid ethereum addresses');
+      });
+
+      it('cannot applyActionToExtensions of creation with refund address not valid', () => {
+        const testnetRefundAddress = deepCopy(MetaCreate.actionCreationFull);
+        testnetRefundAddress.parameters[
+          ExtensionTypes.PAYMENT_NETWORK_ID.ANY_TO_ERC20_PROXY
+        ][0].refundAddress = DataConversionERC20FeeAddData.invalidAddress;
+        // 'must throw'
+        expect(() => {
+          metaPn.applyActionToExtension(
+            MetaCreate.requestStateNoExtensions.extensions,
+            testnetRefundAddress,
+            MetaCreate.requestStateNoExtensions,
+            TestData.otherIdRaw.identity,
+            TestData.arbitraryTimestamp,
+          );
+        }).toThrowError(
+          `refundAddress '${DataConversionERC20FeeAddData.invalidAddress}' is not a valid address`,
+        );
+      });
+      it('keeps the version used at creation', () => {
+        const newState = metaPn.applyActionToExtension(
+          {},
+          { ...MetaCreate.actionCreationFull, version: 'ABCD' },
+          MetaCreate.requestStateNoExtensions,
+          TestData.otherIdRaw.identity,
+          TestData.arbitraryTimestamp,
+        );
+        expect(newState[metaPn.extensionId].version).toBe('ABCD');
+      });
+
+      it('requires a version at creation', () => {
+        expect(() => {
+          metaPn.applyActionToExtension(
+            {},
+            { ...MetaCreate.actionCreationFull, version: '' },
+            MetaCreate.requestStateNoExtensions,
+            TestData.otherIdRaw.identity,
+            TestData.arbitraryTimestamp,
+          );
+        }).toThrowError('version is required at creation');
+      });
+    });
+
+    describe('applyActionToExtension/applyApplyActionToExtension', () => {
```

Comment:
```suggestion

    describe('applyApplyActionToExtension', () => {
```
Just a detail but if you nest it will show the test case path

Reply:
I know, I followed the format of the other tests from this package